Content.Client/RatKing/RatKingSystem.cs

```diff
@@ -1,9 +1,30 @@
-﻿using Content.Shared.RatKing;
+﻿using Content.Shared.Alert.Components;
+using Content.Shared.Nutrition.Components;
+using Content.Shared.RatKing;
 
 namespace Content.Client.RatKing;
 
 /// <inheritdoc/>
 public sealed class RatKingSystem : SharedRatKingSystem
 {
+    public override void Initialize()
+    {
+        base.Initialize();
 
+        SubscribeLocalEvent<RatKingComponent, GetGenericAlertCounterAmountEvent>(OnGetCounterAmount);
+    }
+
+    private void OnGetCounterAmount(Entity<RatKingComponent> ent, ref GetGenericAlertCounterAmountEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (ent.Comp.HungerAlertCategory != args.Alert.Category)
+            return;
+
+        if (!TryComp<HungerComponent>(ent, out var hungerComponent))
+            return;
+
+        args.Amount = (int?)hungerComponent.LastAuthoritativeHungerValue;
```

Comment:
This should use `HungerSystem.GetHunger`. 

The actual hunger value isn't directly tracked but instead calculated from when the hunger was last set from the server (this variable), using a decay rate. This is so that the server doesn't have to constantly update the value on the client.

At the top of the class, include the line
`[Dependency] private readonly HungerSystem _hunger = default!;`
to get the system singleton.